task/download-sbom-from-url-in-attestation/0.1/download-sbom-from-url-in-attestation.yaml

```diff
@@ -0,0 +1,265 @@
+        ]
+      }
+
+      The task will download SBOMs for all .containerImage references.
+    type: string
+
+  - name: SBOMS_DIR
+    default: "."
+    description: >-
+      Path to directory (relative to the 'sboms' workspace) where SBOMs should be downloaded.
+    type: string
+
+  - name: HTTP_RETRIES
+    default: "3"
+    description: "Maximum number of retries for transient HTTP(S) errors"
+    type: string
+
+  - name: PUBLIC_KEY
+    type: string
+    description: >-
+      Public key used to verify signatures. Must be a valid k8s cosign
+      reference, e.g. k8s://my-space/my-secret where my-secret contains
+      the expected cosign.pub attribute.
+    default: ""
+
+  - name: REKOR_HOST
+    type: string
+    description: Rekor host for transparency log lookups
+    default: ""
+
+  - name: IGNORE_REKOR
+    type: string
+    description: >-
+      Skip Rekor transparency log checks during validation.
+    default: "false"
+
+  - name: TUF_MIRROR
+    type: string
+    description: TUF mirror URL. Provide a value when NOT using public sigstore deployment.
+    default: ""
+  workspaces:
+  - name: sboms
+    description: SBOMs will be downloaded to (a subdirectory of) this workspace.
+  stepTemplate:
+    env:
+    - name: IMAGES
+      value: $(params.IMAGES)
+    - name: SBOMS_DIR
+      value: $(workspaces.sboms.path)/$(params.SBOMS_DIR)
+    - name: HTTP_RETRIES
+      value: $(params.HTTP_RETRIES)
+    - name: PUBLIC_KEY
+      value: $(params.PUBLIC_KEY)
+    - name: REKOR_HOST
+      value: $(params.REKOR_HOST)
+    - name: IGNORE_REKOR
+      value: $(params.IGNORE_REKOR)
+    - name: TUF_MIRROR
+      value: $(params.TUF_MIRROR)
+    - name: WORKDIR
+      value: /tekton/home
+  steps:
+  - name: download-attestations
+    image: quay.io/redhat-appstudio/appstudio-utils:5bd7d6cb0b17f9f2eab043a8ad16ba3d90551bc2@sha256:8c7fcf86af40c71aeb58e4279625c8308af5144e2f6b8e28b0ec7e795260e5f7
+    script: |
+      #!/usr/bin/env bash
+      set -o errexit -o nounset -o pipefail
+
+      if [[ -z "$PUBLIC_KEY" ]]; then
+        echo "No public key set, cannot verify attestation." >&2
+        exit 1
+      fi
+
+      cosign_args=(--key "$PUBLIC_KEY")
+
+      if [[ -n "$REKOR_HOST" ]]; then
+        cosign_args+=(--rekor-url "$REKOR_HOST")
+      elif [[ "$IGNORE_REKOR" = "true" ]]; then
+        cosign_args+=(--insecure-ignore-tlog)
+      else
+        cosign_args+=()
+      fi
+
+      if [[ -n "${TUF_MIRROR:-}" ]]; then
+        echo 'Initializing TUF root...'
+        cosign initialize --mirror "${TUF_MIRROR}" --root "${TUF_MIRROR}/root.json"
+      fi
+
+      jq -r '.components[].containerImage' <<< "$IMAGES" | while read -r image; do
+        echo "Getting attestation for $image"
+        mkdir -p "$WORKDIR/$image"
+        cosign verify-attestation \
+          --type slsaprovenance \
+          "${cosign_args[@]}" \
+          "$image" > "$WORKDIR/$image/attestation.json"
+      done
+
+  - name: download-sboms
```

Comment:
I wonder if we should have some functionality in ec itself to download certain artifacts associated with an image.

Reply:
That would certainly make this task less awful. If you feel that downloading SBOMs based on the SBOM_BLOB_URL in the attestation makes sense for the EC CLI, I wouldn't mind that.